### ISSUE TITLE
Implement channel scrollback

### ISSUE THREAD

**Issue by towerofnix** [towerofnix/decent#31]
Scrolling to the top of a channel's message list should load more messages from that channel.

I've already made an API for this: `GET /api/channel/:channelID/latest-messages/before/:messageID`. Message elements from that list should be built the way the elements of the newest messages are, and should be appended above (before) the oldest message element.

When should scrollback message loading occur? It would be easy to make it happen as soon as the screen is scrolled all the way to the top, but maybe fetching and appending messages a little earlier would be better (so as to make scrolling smoother).

When the scrollbar is *dragged* (rather than the scrollwheel being used), should messages only be loaded once the user releases the scrollbar? This way the scrollbar will stay static until it's not immediately being used anymore (so it's easier to navigate to the message you're looking for).

**Comment by heyitsmeuralex** [comment]
The messages cache and API changes in cda18061b934c8c2ceef44462a44734d568e328f should make this a lot easier to do.

**Comment by towerofnix** [comment]
Going to mark this as blocked since fixing this definitely depends on writing the client code, which will only happen after #77 is completed (or merged into master).
